src/EA.Weee.RequestHandlers/Organisations/OrganisationByIdHandler.cs

```diff
@@ -28,17 +29,18 @@ public async Task<OrganisationData> HandleAsync(GetOrganisationInfo query)
         {
             authorization.EnsureInternalOrOrganisationAccess(query.OrganisationId);
 
-            // Need to materialize EF request before mapping (because mapping parses enums)
-            var org = await context.Organisations
-                .SingleOrDefaultAsync(o => o.Id == query.OrganisationId);
+            var org = await context.Organisations.SingleOrDefaultAsync(o => o.Id == query.OrganisationId);
 
             if (org == null)
             {
-                throw new ArgumentException(string.Format("Could not find an organisation with id {0}",
-                    query.OrganisationId));
+                throw new ArgumentException($"Could not find an organisation with id {query.OrganisationId}");
             }
 
-            return organisationMap.Map(org);
+            var organisationData = organisationMap.Map(org);
```

Comment:
Unit test to make sure that mapper is run with the returned organisation

Reply:
Added test